### DISCUSSION TITLE
how to check if geoman editing is enabled

### DISCUSSION THREAD

**Discussion by udos** [geoman-io/leaflet-geoman#1543] [Questions & Help] [answered]
hello,
I would like to prevent saving the geometry while geoman is in "edit" mode (so I would disable the save button while editing) -> what would be a "smart" approach without individually checking the various "edit modes"?

thx 🙏

_note: something like `'pm:globaloptionschanged'` but for when a mode is changed (in which you could check if a mode is active) would be nice :)_

**Comment by Falke-Design** [comment, marked as answer]
The easiest is to check if one of the modes is enabled:
```
const modeEnabled = map.pm.globalCutModeEnabled() || map.pm.globalDragModeEnabled() || map.pm.globalDrawModeEnabled() || map.pm.globalEditModeEnabled() || map.pm.globalRemovalModeEnabled() || map.pm.globalRotateModeEnabled()
```

You can set a flag with listening on the toggle events:
```
let modeEnabled = false;
map.on('pm:globalcutmodetoggled pm:globaldragmodetoggled pm:globaldrawmodetoggled pm:globaleditmodetoggled pm:globalremovalmodetoggled pm:globalrotatemodetoggled ', ()=>{
  modeEnabled = map.pm.globalCutModeEnabled() || map.pm.globalDragModeEnabled() || map.pm.globalDrawModeEnabled() || map.pm.globalEditModeEnabled() || map.pm.globalRemovalModeEnabled() || map.pm.globalRotateModeEnabled()
})
```
